src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs

```diff
@@ -649,17 +661,21 @@ public override int AttributeCount
                     case ScanState.AttrValPseudoValue:
                         return _attrCount;
                     case ScanState.XmlText:
-                        return _textXmlReader.AttributeCount;
+                        {
+                            Debug.Assert(_textXmlReader != null);
+                            return _textXmlReader.AttributeCount;
+                        }
                     default:
                         return 0;
                 }
             }
         }
 
-        public override string GetAttribute(string name, string ns)
+        public override string? GetAttribute(string name, string ns)
         {
             if (ScanState.XmlText == _state)
             {
+                Debug.Assert(_textXmlReader != null);
```

Comment:
I'll leave this as is. In case this is null I'll get test failure (I actually found couple of wrongly annotated cases this way). This is no-op on Release anyway